db/blob/db_blob_basic_test.cc

```diff
@@ -1779,6 +1780,105 @@ TEST_F(DBBlobBasicTest, WarmCacheWithBlobsSecondary) {
             1);
 }
 
+class DBBlobWithTimestampTest : public DBBasicTestWithTimestampBase {
+ protected:
+  DBBlobWithTimestampTest()
+      : DBBasicTestWithTimestampBase("db_blob_with_timestamp_test") {}
+};
+
+TEST_F(DBBlobWithTimestampTest, GetBlob) {
+  Options options = GetDefaultOptions();
+  options.create_if_missing = true;
+  options.enable_blob_files = true;
+  options.min_blob_size = 0;
+  const size_t kTimestampSize = Timestamp(0, 0).size();
+  TestComparator test_cmp(kTimestampSize);
+  options.comparator = &test_cmp;
+
+  DestroyAndReopen(options);
+  WriteOptions write_opts;
+  const std::string ts = Timestamp(1, 0);
+  constexpr char key[] = "key";
+  constexpr char blob_value[] = "blob_value";
+
+  ASSERT_OK(db_->Put(write_opts, key, ts, blob_value));
+
+  ASSERT_OK(Flush());
+
+  Slice read_ts_slice(ts);
```

Comment:
Thanks for looking into this @jowlyzhang ! Based on more recent reports, I think the issue might be still there; could you try reading back the key with a larger timestamp (e.g. `Timestamp(2, 0)`)?

Reply:
That's right. I totally missed the key point. Having a different read timestamp surfaced the issue now. I slightly revised the blob verification logic to ignore timestamp. Let me know if you think this fix is sufficient.

Reply:
I think it would be nice if we could still validate the consistency of the keys exactly (including the timestamp part) so we can detect potential corruptions. I'm wondering if we could use the user key found in the SST file when reading blobs?

Reply:
> This sounds like a perfect plan for me. Do you have any concerns for corner cases that could legitimately fail this consistency check?

Awesome! Re: corner cases, there shouldn't be any: this constraint is supposed to hold in all cases.